src/components/ModalWithForm.js

```diff
@@ -24,4 +27,12 @@ export default class ModalWithForm extends Modal {
       this._handleFormSubmit(this._getInputValues())
     );
   }
+
+  showButtonLoading(text) {
+    this._modalSubmitButton.textContent = text;
+  }
+
+  hideButtonLoading(defaultText) {
+    this._modalSubmitButton.textContent = defaultText;
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```